### PR TITLE
Improve consistency by using response from POST/PUT requests directly to save state

### PR DESCRIPTION
**Pull request by zippolyte**
Avoid consistency issues, due to replication delays, caused by calling the GET endpoint right after POST/PUT

### CODE REVIEW

File: datadog/resource_datadog_role.go [hunk 1]
```diff
@@ -119,102 +120,151 @@ func resourceDatadogRoleCreate(d *schema.ResourceData, meta interface{}) error {
...
-		if permName, ok := permsIDToName[permID]; ok {
-			permR := map[string]string{
-				"id":   permID,
-				"name": permName,
-			}
-			perms = append(perms, permR)
+	switch rolePerms := rolePermsI.(type) {
+	case []datadog.RelationshipToPermissionData:
+		for _, permI := range rolePerms {
+			perms = appendPerm(perms, &permI, permsIDToName)
```
Comment (therve) [inline datadog/resource_datadog_role.go:164]: I think you can just pass `permI.GetId()`, this will simplify the code in `appendPerm`.

**Review by therve — CHANGES_REQUESTED** [review]
One small suggestion, otherwise looks great.

File: datadog/resource_datadog_role.go (hunk 1 above)
Comment (zippolyte) [reply datadog/resource_datadog_role.go:164]: True, thanks for the suggestion

**Review by therve — APPROVED** [review]

**Review by therve — APPROVED** [review]

**Pull request merged**